src/BatteryStateOfChargeService.cpp

```diff
@@ -43,21 +47,49 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalAmpUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (newCurrent_ <= 0)
```

Comment:
If the `newCurrent_` is equal to `0` I would say that it is probably not charging haha.